aspnetcore/fundamentals/host/platform-specific-configuration.md

```diff
@@ -76,7 +76,7 @@ The [sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamen
 * Contains a hosting startup class, `ServiceKeyInjection`, which implements `IHostingStartup`. `ServiceKeyInjection` adds a pair of service strings to the app's configuration using the in-memory configuration provider ([AddInMemoryCollection](/dotnet/api/microsoft.extensions.configuration.memoryconfigurationbuilderextensions.addinmemorycollection)).
 * Includes a `HostingStartup` attribute that identifies the hosting startup's namespace and class.
 
-The `ServiceKeyInjection` class's [Configure](/dotnet/api/microsoft.aspnetcore.hosting.ihostingstartup.configure) method uses an [IWebHostBuilder](/dotnet/api/microsoft.aspnetcore.hosting.iwebhostbuilder) to add enhancements to an app. `IHostingStartup.Configure` in the hosting startup assembly is called by the runtime before `Startup.Configure` in user code, which allows user code to overwrite any configuration provided by the hosting startup assembly.
+The `ServiceKeyInjection` class's [Configure](/dotnet/api/microsoft.aspnetcore.hosting.ihostingstartup.configure) method uses an [IWebHostBuilder](/dotnet/api/microsoft.aspnetcore.hosting.iwebhostbuilder) to add enhancements to an app. `IHostingStartup.Configure` in the hosting startup assembly is called by the runtime before `Startup.Configure` in user code. However, delegates registered by `IHostingStartup` (for example, when calling `ConfigureAppConfiguration`) run after user code. To ensure that configuration code that relies on callbacks is executed before user code, call `UseConfiguration` to apply the configuration to the app.
```

Comment:
I think the really confusing thing here is that it's comparing `IHostingStartup.Configure(IWebHostBuilder)` to `Startup.Configure(IApplicationBuilder)`. The two aren't comparable. I'm not sure why the comment "which allows user code to overwrite any configuration provided by the hosting startup assembly" was ever there, Startup.Configure can't override anything on the IWebHostBuilder. Maybe this whole paragraph should be removed.

Reply:
> not sure why the comment ... was ever there

I can research that, but I have a feeling that it will just come back as my blowing a comment based on an engineering issue/PR comment that review didn't catch.

I think "any configuration" was supposed to mean "app configuration" (configuration-provider-based configuration), and then it looks like the remark is **wrong** anyway based on the issue discussion (delegates from `IHostingStartup` go **after** those from the app).

It will probably help me to repro what the OP said was happening. I usually write a better bit when I see it with my own :eyes:.

I'll try again on Thursday morning. :coffee:

Reply:
The config ordering is a real issue, but that has nothing to do with Startup.Configure. If you want IHostingStartup to add config before the application's ConfigureAppConfiguration then you need to call UseConfiguration which executes immediately rather than as a delayed callback.